src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/DirichletAnalytic.cpp

```diff
@@ -26,6 +51,81 @@ DirichletAnalytic<Dim>::get_clone() const {
 template <size_t Dim>
 void DirichletAnalytic<Dim>::pup(PUP::er& p) {
   BoundaryCondition<Dim>::pup(p);
+  p | analytic_prescription_;
+}
+
+template <size_t Dim>
+std::optional<std::string> DirichletAnalytic<Dim>::dg_ghost(
+    const gsl::not_null<tnsr::aa<DataVector, Dim, Frame::Inertial>*>
+        spacetime_metric,
+    const gsl::not_null<tnsr::aa<DataVector, Dim, Frame::Inertial>*> pi,
+    const gsl::not_null<tnsr::iaa<DataVector, Dim, Frame::Inertial>*> phi,
+    const gsl::not_null<Scalar<DataVector>*> gamma1,
+    const gsl::not_null<Scalar<DataVector>*> gamma2,
+    const gsl::not_null<Scalar<DataVector>*> lapse,
+    const gsl::not_null<tnsr::I<DataVector, Dim, Frame::Inertial>*> shift,
+    const gsl::not_null<tnsr::II<DataVector, Dim, Frame::Inertial>*>
+        inv_spatial_metric,
+    const std::optional<
+        tnsr::I<DataVector, Dim, Frame::Inertial>>& /*face_mesh_velocity*/,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& /*normal_covector*/,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& /*normal_vector*/,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& coords,
+    const Scalar<DataVector>& interior_gamma1,
+    const Scalar<DataVector>& interior_gamma2, const double time) const {
+  *gamma1 = interior_gamma1;
+  *gamma2 = interior_gamma2;
+  ASSERT(analytic_prescription_ != nullptr,
+         "The analytic prescription must be set.");
+  auto boundary_values = call_with_dynamic_type<
+      tuples::TaggedTuple<
+          GeneralizedHarmonic::Tags::Pi<Dim, Frame::Inertial>,
+          GeneralizedHarmonic::Tags::Phi<Dim, Frame::Inertial>,
+          gr::Tags::SpacetimeMetric<Dim, Frame::Inertial, DataVector>>,
+      tmpl::append<
+          GeneralizedHarmonic::Solutions::all_solutions<Dim>,
+          tmpl::conditional_t<
+              Dim == 3,
+              tmpl::append<
+                  GeneralizedHarmonic::Solutions::RelativisticEuler::
+                      all_solutions,
+                  GeneralizedHarmonic::Solutions::grmhd::all_solutions,
+                  GeneralizedHarmonic::AnalyticData::grmhd::all_analytic_data>,
```

Comment:
This list of solutions has shown up twice now (first time in GhValenciaDivCleanBase with Dim==3). Maybe factor it out somewhere so it can be reused?